### PR TITLE
Fix restore.cmd/sh not recognizable command

### PR DESCRIPTION
**Pull request by schellap**
@davidfowl, PTAL. Fixing the Issue #186 for restore.cmd. Add on to
PR #196.

@anurse, @livarcocc, @krwq

### CODE REVIEW

**Comment by schellap** [comment]
@gkhanna79, this is the issue due to which my CI was failing: https://github.com/dotnet/corert/pull/404

File: src/Microsoft.DotNet.Cli.Utils/Command.cs [hunk 1]
```diff
@@ -57,11 +57,18 @@ public static Command Create(string executable, string args)
 
         private static void ResolveExecutablePath(ref string executable, ref string args)
         {
-            var fullExecutable = Path.GetFullPath(Path.Combine(AppContext.BaseDirectory, executable + Constants.ExeSuffix));
-
-            if (File.Exists(fullExecutable))
+            foreach (string suffix in Constants.RunnableSuffixes)
```
Comment (davidfowl) [inline src/Microsoft.DotNet.Cli.Utils/Command.cs]: This seems like it shouldn't be required. On *nix the file should be marked executable. On windows the `ShouldUseCmd` method should handle running `.cmd` files.

File: src/Microsoft.DotNet.Cli.Utils/Constants.cs [hunk 2]
```diff
@@ -12,6 +12,12 @@ namespace Microsoft.DotNet.Cli.Utils
     internal static class Constants
     {
         public static readonly string ExeSuffix = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? ".exe" : string.Empty;
+
+        // Priority order of runnable suffixes to look for and run
+        public static readonly string[] RunnableSuffixes = RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
+                                                         ? new string[] { ".cmd", ".bat", ".exe" } 
+                                                         : new string[] { ".sh", string.Empty };
```
Comment (davidfowl) [inline src/Microsoft.DotNet.Cli.Utils/Constants.cs]: `Array.Empty<string>()` for non windows.

File: src/Microsoft.DotNet.Cli.Utils/Command.cs (hunk 1 above)
Comment (schellap) [inline src/Microsoft.DotNet.Cli.Utils/Command.cs]: We discussed this and the summary is ShouldUseCmd does a no-op for this case, hence the bug.

File: src/Microsoft.DotNet.Cli.Utils/Constants.cs (hunk 2 above)
Comment (schellap) [inline src/Microsoft.DotNet.Cli.Utils/Constants.cs]: We need to have one entry at least, the basic exe, to be backwards compatible with original code on Unix.

File: src/Microsoft.DotNet.Cli.Utils/Constants.cs [hunk 3]
```diff
@@ -12,6 +12,12 @@ namespace Microsoft.DotNet.Cli.Utils
     internal static class Constants
     {
         public static readonly string ExeSuffix = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? ".exe" : string.Empty;
+
+        // Priority order of runnable suffixes to look for and run
+        public static readonly string[] RunnableSuffixes = RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
+                                                         ? new string[] { ".cmd", ".bat", ".exe" }
```
Comment (davidfowl) [inline src/Microsoft.DotNet.Cli.Utils/Constants.cs]: I'd change the order to look for exe first.

**Comment by davidfowl** [comment]
Minor nit then :shipit:

File: src/Microsoft.DotNet.Cli.Utils/Constants.cs (hunk 3 above)
Comment (schellap) [inline src/Microsoft.DotNet.Cli.Utils/Constants.cs]: Are you sure? Usually people have customization on top of the .exe in a .cmd/bat file to specify default arguments etc.

**Pull request merged**